src/mtev_conf.c

```diff
@@ -1100,21 +1100,50 @@ static int mtev_conf_check_value(mtev_conf_description_t* description) {
   return rv;
 }
 
+char*
+mtev_conf_section_to_xpath(mtev_conf_section_t* section) {
+  if (section == NULL) {
+    return NULL;
+  }
+
+  const int max_len = 512;
+  char* string = alloca(max_len);
+  char* string_ptr = string + max_len;
+
+  xmlNode* node = (xmlNode*) section;
+  while (node && node->name) {
+    int len = strlen(node->name);
+    string_ptr -= len+1;
+    if(string_ptr < string) {
+      break;
+    }
+    memcpy(string_ptr+1, node->name, len);
```

Comment:
/foo/bar/baz isn't an unambiguous XPath name.  Think of documents with multiple bar and multiple baz nodes.